Self-Rewarding Algorithm with TRT Support (#321)
NVIDIA/NeMo-Aligner: docs/user-guide-experimental/generation.rst

The following block shows the standard Nemo sampling params for generating responses, which are the same as we use across all Nemo and Aligner codebases:

.. code-block:: yaml
    sampling_params:
      use_greedy: False
      temperature: 1.0
      top_k: 0
      top_p: 1.0
      repetition_penalty: 1.0
      add_BOS: False
      all_probs: False
      compute_logprob: False
      end_strings: ["<|endoftext|>", "<extra_id_1>"]

    # length argument for autoregressive sampling
    # max length means max amount of tokens to generate
    length_params:
      max_length: ${int_div:${model.encoder_seq_length}, 2}
      min_length: 1

Finally, we have the TRT parameters, which allows for faster TRTLLM-based response generation:

.. code-block:: yaml
    trt_llm:
      enable: True  # use this to turn TRT on/off
      # reshard: False # reshard is not supported in generation

      # TRTLLM preallocates activation memory according to the number of input tokens
      max_input_len: ${subtract:${model.encoder_seq_length}, ${model.generation.length_params.max_length}}

      model_type: gptnext # can be gptj, gptnext, llama, gemma, falcon

>       # Save GPU memory by unloading and reloading the TRTLLM engine before and after the training stage
>       # Reloading the engine incurs a constant time overhead
>       unload_engine_train: False

Reviewer: ```suggestion
      # Generation does not have a training stage, so there is no need to unload the engine.
      unload_engine_train: False
```
